samples/file-encryptor/host/host.cpp

```diff
@@ -266,16 +269,22 @@ int encrypt_file(
             goto exit;
         }
 
+        bytes_to_write = bytes_read;
```

Comment:
Can you elaborate a bit on what the bug is? It seems that original logic around `bytes_left` already handles the last block. If the `file_data_size` is less than the `DATA_BLOCK_SIZE`, doesn't the `bytes_read` returned by `fread` equals to `file_data_size`?

Reply:
The size of testfile is 794 bytes and that of out.encrypted file is 800 bytes (with 6 bytes of padding).  When encrypting, we add  6 bytes of padding bytes to the testfile, to get block size that works out for CBC. 

In previous implementation when decrypting, we were taking in 800 bytes of out.encrypted and decrypting it to get 800 bytes of out.decrypted. The last 6 bytes were the padding bytes in this file. We shouldn't be writing the padding bytes to the out.decrypted file. In this change, I am writing only the first 794 bytes (hence the need for data_left).

bytes_left -> initialized to size of data in out.encrypted
data_left -> initialized to size of data in testfile (from header.file_data_size)

Reply:
Checkout the last few bytes in testfile and out.decrypted files in before (1st execution) and after (2nd execution) the change is applied.
![out decrypted results](https://user-images.githubusercontent.com/20543961/109077438-ff087c80-76b0-11eb-8190-b5363cb1194c.png)

Reply:
It seems to me that the bug might just be here
https://github.com/openenclave/openenclave/blob/f1f9635e0c2a00d69210ab7f259b00052124cf52/samples/file-encryptor/host/host.cpp#L190
Would change this to
```
src_data_size = header.file_data_size; 
```
help?

Reply:
So the last block of out.encrypted file needs to be passed in with the extra bytes added for padding to encrypt_block() (as these are needed to retrieve the initial data)...